app/src/test/java/com/github/multimatum_team/multimatum/DisplayLocationActivityTest.kt

```diff
@@ -0,0 +1,123 @@
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import javax.inject.Singleton
+
+@UninstallModules(DisplayLocationActivityModule::class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4::class)
+class DisplayLocationActivityTest {
+
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @Before
+    fun init() {
+        Intents.init()
+        hiltRule.inject()
+    }
+
+    @After
+    fun release() {
+        Intents.release()
+    }
+
+    @Test
+    fun `initializeMapView should initialize the mapView correctly`() {
+
+        val mockMapBoxMap: MapboxMap = mock()
+        val mockAnnotationsPlugin: AnnotationPlugin = mock()
+        val mockPointAnnotationManager: PointAnnotationManager = mock()
+
+        var cameraOptions: CameraOptions? = null
+        var pointAnnotationManagerHasBeenCreated = false
+
+        // Setup the mocks to set the variables with the values they are given as arguments
+        // and to return other mocks when needed
+        whenever(mockMapView.getMapboxMap()).thenReturn(mockMapBoxMap)
+        whenever(mockMapBoxMap.loadStyleUri(anyString(), any<Style.OnStyleLoaded>())).then {
+            val onStyleLoaded: Style.OnStyleLoaded = it.getArgument(1)
+            onStyleLoaded.onStyleLoaded(mock())
+        }
+        whenever(mockMapBoxMap.setCamera(any<CameraOptions>())).then {
+            cameraOptions = it.getArgument(0)
+            null
+        }
+        // mocking getPlugin instead of annotations because it is a delegate property (and calls getPlugin)
+        whenever(mockMapView.getPlugin<AnnotationPlugin>(Plugin.MAPBOX_ANNOTATION_PLUGIN_ID)).thenReturn(mockAnnotationsPlugin)
+        whenever(mockAnnotationsPlugin.createPointAnnotationManager()).thenReturn(
+            mockPointAnnotationManager
+        )
+        whenever(mockPointAnnotationManager.create(any<PointAnnotationOptions>())).then {
+            pointAnnotationManagerHasBeenCreated = true
+            mock<PointAnnotation>()
+        }
+
+        val applicationContext = ApplicationProvider.getApplicationContext<Context>()
+        val intent = Intent(applicationContext, DisplayLocationActivity::class.java)
+        val activityScenario: ActivityScenario<DisplayLocationActivity> =
+            ActivityScenario.launch(intent)
+
+        activityScenario.use {  }
+
+        assertNotNull(cameraOptions)
+        assertEquals(14.0, cameraOptions!!.zoom)
+        assertTrue(pointAnnotationManagerHasBeenCreated)
+
+    }
+
+    companion object {
+        private val mockMapView: MapView = mock()
+    }
+
+    @Module
+    @InstallIn(SingletonComponent::class)
+    object TestDependencyProvider {
+
+        @Singleton
+        @Provides
+        fun provideMapViewProducer(): MapViewProducer =
+            MapViewProducer { mockMapView }
+
+        @Singleton
+        @Provides
+        fun provideContentViewSetter(): ViewActionPerformer =
```

Comment:
Are the two provides logically coupled? When I put every repository in the FirebaseRepositoryModule, it turned out to be inconvenient in tests since I had to provide all repositories in the test module even when they were not used. @LennyDZ and I had to update almost all the tests everytime we added a new type of repository.